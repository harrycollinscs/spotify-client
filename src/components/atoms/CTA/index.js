import React from "react";
import styled from "styled-components";

export const StyledButton = styled.button`
  background-color: ${props => props.primary ? '#1db954' : '#212121' };
  width: 200px;
  height: 50px;
  border-radius: 50px;
  border-style: none;
  color: white;
`;

const CTA = ({ children, action, primary = true }) => <StyledButton primary={primary} onClick={action}>{children}</StyledButton>;

export default CTA;
