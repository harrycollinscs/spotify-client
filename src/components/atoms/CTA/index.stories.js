import React from "react";
import CTA from "./index";

export default {
  title: "Atoms/CTA",
  component: CTA,
  argTypes: {
    action: {
      action: "clicked",
    },
    primary: {
      options: [true, false],
      control: { type: "radio" },
    },
  },
};

const Template = (args) => <CTA {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  children: "Click me!",
  primary: true,
};

export const Secondary = Template.bind({});
Secondary.args = {
  children: "Click me!",
  primary: false,
};
