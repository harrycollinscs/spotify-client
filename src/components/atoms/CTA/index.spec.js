import React from "react";
import CTA, { StyledButton } from "./index";
import { shallow } from "enzyme";

describe("Atoms/CTA", () => {
  const props = {
    primary: true,
    action: jest.fn(),
  };

  const component = <CTA {...props}>Click me!</CTA>;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });

  it("Should call action on click", () => {
    const render = shallow(component);
    render.find(StyledButton).simulate("click");
    expect(props.action).toHaveBeenCalled();
  });
});
