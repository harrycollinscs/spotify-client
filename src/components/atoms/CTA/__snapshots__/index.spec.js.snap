// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/CTA Should match snapshot 1`] = `
<styled.button
  onClick={[MockFunction]}
  primary={true}
>
  Click me!
</styled.button>
`;
